### DISCUSSION TITLE
Unable to run the model, please provide solutions,  newcomer

### DISCUSSION THREAD

**Discussion by xieyuchengwin** [open-compass/opencompass#797] [Q&A]
运行python run.py --models hf_llama_7b --datasets mmlu_ppl ceval_ppl

会报错gpus为[],
<img width="1685" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/5509d36c-66e8-41f2-9c06-d2abe3740169">

  0%|                                                                                                                                                                              | 0/4 [00:00<?, ?it/s]len(gpus),gpus,num_gpus 0 [] 1
len(gpus),gpus,num_gpus 0 [] 1
len(gpus),gpus,num_gpus 0 [] 1
len(gpus),gpus,num_gpus 0 [] 1
Traceback (most recent call last):
  File "/Users/xieyucheng/Documents/python_projects/opencompass/run.py", line 357, in <module>
    main()
  File "/Users/xieyucheng/Documents/python_projects/opencompass/run.py", line 309, in main
    runner(tasks)
  File "/Users/xieyucheng/Documents/python_projects/opencompass/opencompass/runners/base.py", line 39, in __call__
    self.summarize(status)
  File "/Users/xieyucheng/Documents/python_projects/opencompass/opencompass/runners/base.py", line 61, in summarize
    for _task, code in status:
  File "/Users/xieyucheng/anaconda3/envs/opencompass/lib/python3.10/concurrent/futures/_base.py", line 621, in result_iterator
    yield _result_or_cancel(fs.pop())
  File "/Users/xieyucheng/anaconda3/envs/opencompass/lib/python3.10/concurrent/futures/_base.py", line 319, in _result_or_cancel
    return fut.result(timeout)
  File "/Users/xieyucheng/anaconda3/envs/opencompass/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/Users/xieyucheng/anaconda3/envs/opencompass/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/Users/xieyucheng/anaconda3/envs/opencompass/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/xieyucheng/Documents/python_projects/opencompass/opencompass/runners/local.py", line 123, in submit
    assert len(gpus) >= num_gpus
AssertionError
  0%|

**Comment by xieyuchengwin** [comment]
其他运行命令没有看到问题
<img width="1213" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/022fb453-8458-4ad3-a14c-53e920ba8f65">

**Comment by bittersweet1999** [comment]
It seems that you are running OpenCompass on a MacBook. Please try specifying the GPU numbers and ensure that your GPU is available.

**Reply by xieyuchengwin** [reply]
thanks，i have a question，i run mambaforge ,GPU is available, conda is run.

**Reply by xieyuchengwin** [reply]
please help me，
https://github.com/open-compass/opencompass/discussions/797#discussioncomment-8132364

**Reply by Pandoroda** [reply]
我也出现了这样的问题 macbook m1pro的电脑，请问你是怎么解决的呀，我查了一下macos自己的芯片没有支持版本的cuda了，缺失cuda所以跑不了

**Comment by tonysy** [comment]
Make sure the pytorch-cuda is valid on your mac. I'm not sure the current version is valid for mac os or not.

**Reply by xieyuchengwin** [reply]
thanks, i give up mac，try run linux

**Reply by xieyuchengwin** [reply]
hallo,我使用centos服务器运行一样不可以，可以帮助我解决问题吗？
<img width="1662" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/f9672424-d0bf-4ae6-8b96-e87af68ea2de">

**Reply by xieyuchengwin** [reply]
please help me，
https://github.com/open-compass/opencompass/discussions/797#discussioncomment-8132364

**Reply by xieyuchengwin** [reply]
张老师，能帮我一下吗？

**Comment by xieyuchengwin** [comment]
我服务器运行命令如下（use centOS7），please help me：
1、install Miniconda
2、Open-source Models with GPU or API Models with CPU-only
<img width="1006" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/1eb9826d-dc6b-4fc9-9c0e-c2da45087c8d">
<img width="998" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/88b4019a-10d6-4fd0-a8de-6da2c2c9e631">
<img width="651" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/2bee1122-74a5-4084-8858-cae62ba16518">
<img width="496" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/d63f2b4d-4dc4-45b5-ace5-28a67f906120">
3、Download dataset to data/ folder
<img width="717" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/af01bbbd-f892-4e04-8b21-3e6ba641fddf">
4、run python
python run.py --models hf_llama_7b --datasets mmlu_ppl ceval_ppl --debug
<img width="1663" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/a3b4adf2-19ba-4a68-838d-ff80f65dd214">

**Comment by tonysy** [comment]
请确认你的pytorch是cuda版本的么？看起来没有成功装上cuda版本的pytorch

**Reply by xieyuchengwin** [reply]
<img width="877" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/9af04722-7f40-4566-9bcb-f5e3ae79c130">
版本为12.1

**Reply by tonysy** [reply]
```bash
import torch
torch.cuda.is_available()
```

**Reply by bittersweet1999** [reply]
try nvidia-smi and torch.cuda.is_available() ?

**Reply by xieyuchengwin** [reply]
<img width="494" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/b3ab48d1-858d-4218-b213-d4c2651359d4">
False

**Reply by bittersweet1999** [reply]
use nvcc -V to see your cuda version, maybe there is some confict between the cuda version and torch version

**Reply by xieyuchengwin** [reply]
nvidia-smi ：
<img width="669" alt="图片" src="https://github.com/open-compass/opencompass/assets/52005561/16511a70-36b3-42ae-8679-6cfd8808b29b">

**Reply by tonysy** [reply]
Make sure you have installed nvidia driver, and install the cuda version pytorch   correctly.

**Reply by xieyuchengwin** [reply]
我只能使用nvidia的显卡吗？

**Reply by xieyuchengwin** [reply]
张老师，我已经可以跑起来了，但是结果有些问题，请问怎么解决
![图片](https://github.com/open-compass/opencompass/assets/52005561/0a1945f3-3039-4930-868f-0481431b7102)
![图片](https://github.com/open-compass/opencompass/assets/52005561/21f23522-7d14-47a9-a535-b6da858cc34e)

failed to create process.
